math/mathcore/inc/Fit/EvaluateLogL.hxx

```diff
@@ -0,0 +1,569 @@
+// @(#)root/mathcore:$Id$
+// Author: L. Moneta Tue Nov 28 10:52:47 2006
+
+/**********************************************************************
+ *                                                                    *
+ * Copyright (c) 2006  LCG ROOT Math Team, CERN/PH-SFT                *
+ *                                                                    *
+ *                                                                    *
+ **********************************************************************/
+
+#ifndef ROOT_Fit_EvaluateLogL
+#define ROOT_Fit_EvaluateLogL
+
+#include "Math/IParamFunctionfwd.h"
+#include "Math/IParamFunction.h"
+
+#ifdef R__USE_IMT
+#include "ROOT/TThreadExecutor.hxx"
+#endif
+#include "ROOT/TSequentialExecutor.hxx"
+
+#include "Fit/BinData.h"
+#include "Fit/UnBinData.h"
+#include "Fit/FitExecutionPolicy.h"
+#include "Fit/FitUtil.h"
+
+#include "Math/Integrator.h"
+#include "Math/IntegratorMultiDim.h"
+
+#include "TError.h"
+
+// using parameter cache is not thread safe but needed for normalizing the functions
+#define USE_PARAMCACHE
+
+
+namespace ROOT {
+
+namespace Fit {
+
+/**
+   namespace defining utility free functions using in Fit for evaluating the various fit method
+   functions (chi2, likelihood, etc..)  given the data and the model function
+
+   @ingroup FitMain
+*/
+namespace FitUtil {
+
+  // internal class defining
+  template <class T>
+  class LikelihoodAux {
+  public:
+     LikelihoodAux(T logv = {}, T w = {}, T w2 = {}) : logvalue(logv), weight(w), weight2(w2) {}
```

Comment:
If you use `T(0)`, you may not need the specialization below, which is just code duplication that is error prone (you may edit here and forget to update below.